lib/csv/table.rb

```diff
@@ -285,7 +301,8 @@ def delete_if(&block)
         @table.delete_if(&block)
       else                                      # by header
         deleted = []
-        headers.each do |header|
+        initial_headers = headers
```

Comment:
I can see you want to iterate through the initial headers of the table, but at this line `headers` is equals `@headers.dup`. Looks like we don't need this extra assignment.

Reply:
You're right, I've fixed it.